airflow/jobs/triggerer_job_runner.py

```diff
@@ -608,6 +609,17 @@ async def run_trigger(self, trigger_id, trigger):
                 self.log.info("Trigger %s fired: %s", self.triggers[trigger_id]["name"], event)
                 self.triggers[trigger_id]["events"] += 1
                 self.events.append((trigger_id, event))
+        except asyncio.CancelledError as err:
+            trigger_timeout = trigger.task_instance.trigger_timeout
+            if trigger_timeout:
+                if not trigger_timeout.tzinfo:
+                    trigger_timeout = trigger_timeout.replace(tzinfo=timezone.utc)
+
+                if trigger_timeout < timezone.utcnow():
+                    self.log.error("Trigger cancelled due to timeout")
+
+            self.log.error("Trigger cancelled; message=%s", err)
+            raise
```

Comment:
this suggestion is a bit golfy, but hey

```suggestion
        except asyncio.CancelledError as err:
            if timeout := trigger.task_instance.trigger_timeout:
                timeout = timeout.replace(tzinfo=timezone.utc) if not timeout.tzinfo else timeout
                if timeout < timezone.utcnow():
                    self.log.error("Trigger cancelled due to timeout")
            self.log.error("Trigger cancelled; message=%s", err)
            raise
```

Reply:
Thanks for your suggestion! It looks great!